configs/example/gem5_library/checkpoints/simpoints-se-checkpoint.py

```diff
@@ -0,0 +1,70 @@
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+from gem5.components.boards.simple_board import SimpleBoard
+from gem5.components.memory import DualChannelDDR4_2400
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.isas import ISA
+from gem5.resources.resource import Resource
+from pathlib import Path
+from gem5.components.cachehierarchies.classic.\
+    private_l1_private_l2_cache_hierarchy import(
+    PrivateL1PrivateL2CacheHierarchy
+)
+
+from gem5.components.processors.simpoint import SimPoint
+from gem5.simulate.exit_event_generators import (
+    simpoint_save_checkpoint_generator,
+)
+
+requires(isa_required=ISA.X86)
+
+cache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
```

Comment:
I suggest using `NoCache` when taking checkpoints for SimPoints (at least in our examples). When restoring from a checkpoint, there's no cache state saved, so you can completely change the cache hierarchy. By using `NoCache` to take the checkpoints, we'll get slightly better performance when running in atomic mode, and it won't put any restrictions on what people can do *with* the checkpoints. 

As an aside, when we finalize these scripts as examples, we should write things like the above in comments so everyone can learn.

Reply:
I'll change to NoCache and add the comments. It's a very useful information. Thank you!